### ISSUE TITLE
Faulty unicode conversion when saving ods in environments with ASCII as default encoding

### ISSUE THREAD

**Issue by felixring** [chfw/pyexcel-ods#2]
Hi,

I ran into the beloved
> UnicodeEncodeError: 'ascii' codec can't encode character u'\xd6' in position 1: ordinal not in range(128)

when saving data containing non ASCII characters to ods.

Python 2.6.6 on CentOS 6, kernel  2.6.32-504.16.2.el6.x86_64
pyexcel version: 0.1.7

A specific test code that triggers the problem is the following:
```python
#!/usr/bin/python
# -*- encoding: utf-8 -*-
import pyexcel as pe
from pyexcel.ext import ods

data = [[1,2,3],['Joe', unicode('DÖE', 'utf-8', 'strict'), 'Name']]
my_sheet = pe.Sheet(data)
my_sheet.save_as('test.ods')
```

A quick analysis showed that the type is recognized correctly, but somehow the conversion gets called twice when setting the attributes, once for the value-type, which specifies the type correctly as string, and then again for the value, which uses cnv_double, which breaks because of the non ASCII characters. I stopped debugging from there, as I would need a more thorough understanding of the employed conversion mechanisms.

Here is the full traceback of the error:
> Traceback (most recent call last):
  File "odstest.py", line 8, in <module>
    my_sheet.save_as('test.ods')
  File "/usr/lib/python2.6/site-packages/pyexcel/sheets/sheet.py", line 39, in save_as
    return self.save_to(source)
  File "/usr/lib/python2.6/site-packages/pyexcel/sheets/sheet.py", line 33, in save_to
    source.write_data(self)
  File "/usr/lib/python2.6/site-packages/pyexcel/sources/file.py", line 35, in write_data
    w.write_reader(sheet)
  File "/usr/lib/python2.6/site-packages/pyexcel/writers.py", line 75, in write_reader
    self.write_rows(to_array(reader))
  File "/usr/lib/python2.6/site-packages/pyexcel/writers.py", line 45, in write_rows
    self.writer.write_array(table)
  File "/usr/lib/python2.6/site-packages/pyexcel_io/base.py", line 222, in write_array
    self.write_row(r)
  File "/usr/lib/python2.6/site-packages/pyexcel_ods/__init__.py", line 257, in write_row
    self.write_cell(tr, x)
  File "/usr/lib/python2.6/site-packages/pyexcel_ods/__init__.py", line 246, in write_cell
    tc.setAttrNS(OFFICENS, x_odf_value_token, x)
  File "/usr/lib/python2.6/site-packages/odf/element.py", line 440, in setAttrNS
    self.attributes[(namespace, localpart)] = c.convert((namespace, localpart), value, self)
  File "/usr/lib/python2.6/site-packages/odf/attrconverters.py", line 1489, in convert
    return conversion(attribute, value, element)
  File "/usr/lib/python2.6/site-packages/odf/attrconverters.py", line 77, in cnv_double
    return str(arg)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xd6' in position 1: ordinal not in range(128)

and here is the output of an added print statement in attrconverters.py:
```python
print attribute, value, element
```
> (u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', u'mimetype') application/vnd.oasis.opendocument.spreadsheet
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', u'version') 1.1
(u'urn:oasis:names:tc:opendocument:xmlns:table:1.0', u'name') pyexcel
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value-type') float
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value') 1
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value-type') float
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value') 2
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value-type') float
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value') 3
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value-type') string
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value') Joe
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value-type') string
(u'urn:oasis:names:tc:opendocument:xmlns:office:1.0', 'value') DÖE

I would greatly appreciate your help on solving this issue.
Thanks for the awesome package anyways.

Best regards,
Felix

**Comment by chfw** [comment]
Issue was reproduced. I will find a solution to it.

**Comment by chfw** [comment]
while I am seeking the solution, you may use its sister library: pyexcel-ods3:

The following code works OK: 

    #!/usr/bin/python
    # -*- encoding: utf-8 -*-
    import pyexcel as pe
    from pyexcel.ext import ods3

    data = [[1,2,3],['Joe', unicode('DÖE', 'utf-8', 'strict'), 'Name']]
    my_sheet = pe.Sheet(data)
    my_sheet.save_as('test.ods')

**Comment by chfw** [comment]
Please try it out by:

    pip install git+https://github.com/chfw/pyexcel-ods.git

**Comment by chfw** [comment]
released to pypi as 0.0.9. pip shall be able to get it on fresh installation. for existing one, use the following commands:


     pip install pyexcel-ods==0.0.9

or

    pip install --upgrade pyexcel-ods

reopen this issue if it is not fixed for you.

**Comment by felixring** [comment]
Thanks for the quick response. Works perfectly now!
